src/components/canvases/CanvasesWrapper.tsx

```diff
@@ -0,0 +1,30 @@
+import { ReactNode } from "react";
+import { useCanvasElements } from "../../hooks/useMutatedElements";
+import { CommonCanvasAppState } from "../../types";
+import { NonDeletedExcalidrawElement } from "../../element/types";
+import Scene from "../../scene/Scene";
+import { useVisibleCanvasElements } from "../../hooks/useVisibleElements";
+
+type CanvasesWrapperProps = {
+  appState: CommonCanvasAppState;
+  scene: Scene;
+  children: (
+    versionNonce: number | undefined,
+    elements: readonly NonDeletedExcalidrawElement[],
+    visibleElements: readonly NonDeletedExcalidrawElement[],
+  ) => ReactNode;
+};
+
+const CanvasesWrapper = ({
```

Comment:
The wrapper would not be realy needed if elements and visible elements were provided by the Scene (and cached). Decided not to change it in this PR since there are already too many changes.